extensions/logs/suite.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type LogSuite struct {
+	LogDir string
+
+	testStartTime time.Time
+	kubeClient    kubernetes.Interface
+}
+
+func (s *LogSuite) SetupSuite() {
+	if s.LogDir == "" {
+		s.LogDir = "./logs"
+	}
+
+	kubeClient, err := newKubeClient()
+	if err != nil {
+		panic(err)
+	}
+
+	s.kubeClient = kubeClient
+}
+
+func (s *LogSuite) SetupTest() {
+	s.testStartTime = time.Now()
+}
+
+func (s *LogSuite) AfterTest(suiteName, testName string) {
+	logOptions := corev1.PodLogOptions{
+		Timestamps: true,
+		SinceTime:  &metav1.Time{Time: s.testStartTime},
+	}
+
+	logPath := fmt.Sprintf("%s/%s/%s", s.LogDir, suiteName, testName)
+	if err := os.MkdirAll(logPath, os.ModePerm); err != nil {
+		panic(err)
+	}
+
+	list, _ := s.kubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
Please do not use `context.TODO()` and `context.Background()`, lets use context with timeouts.

Reply:
Done